### DISCUSSION TITLE
How to build

### DISCUSSION THREAD

**Discussion by cdavidyoung** [shinyorg/shiny#860] [Q&A]
I am looking for a BLE framework for a mobile application and this looks interesting.  However, when I build the shiny solution I get many errors with the latest VS 2022.  Would it be possible to provide some instructions to get started?

Thanks

**Comment by aritchie** [comment]
There isn't any special build instructions.  If you are getting errors, you likely don't have the right workloads for VS installed.  I'm not sure why you wouldn't just use the nuget packages instead?

**Comment by cdavidyoung** [comment]
Ok, fair enough.  When I go to Nuget Package Manager and browse for shiny I
get a lot of packages.  I hate to sound like a total newbie but which ones
do I need?  I installed Shiny, Shiny.Core, and Shiny.BluetoothLE and my
empty project compiles, at least the iOS target.  The Android gets an error
but I unloaded it for now.

Presumably some of the sample projects would build but I am not sure.  I
would like to start with some working Xamarin Forms BLE project and go from
there.

I have a working peripheral that is sending beacons and will connect with a
standard BLE app on my phone.  It sends out strings via the UART service
that I wish to parse with my mobile app and store in a database.  I am also
sending gps coordinates back to the peripheral that it includes in its UART
string.

Any pointers would be appreciated.  If I am successful at getting this
going I would be happy to write up instructions that could be included in
the git repo for newbies like me.

Thanks
Charles

**Comment by cdavidyoung** [comment]
Allan,

I read through your blog at allanritchie.com and this looks like impressive
work.

So I created a new mobile Xamarin forms project and added the Shiny.Beacons
nuget, hoping that it would be a working project.  It seems that I need to
add some code to my project to actually get it to do something though.

Am I missing something?  Are there really no sample Shiny BLE projects that
will build and do something even if it is something so simple as scanning
for beacons?

Thanks
Charles

**Reply by aritchie** [reply]
Nuget is for libraries.  It doesn't add a full working app or sample.  Please refer to the readme on this project as it has links to some samples (I'm still working on them) but the beacon sample is there.

**Comment by cdavidyoung** [comment]
I had tried this before.  This is what I get when I build the Beacons
sample.

Error CS8370 Feature 'nullable reference types' is not available in C# 7.3.
Please use language version 8.0 or greater. Sample.Shared
C:\Users\charl\source\repos\shiny\_Template\Sample.Shared\Infrastructure\CommandItem.cs
9 Active

I tried changing the .csproj file to one suggested solution but it made no
difference:

<Project Sdk="Microsoft.NET.Sdk">

<PropertyGroup>
<TargetFramework>netstandard3.1</TargetFramework>
                <LangVersion>8.0</LangVersion>
<ProduceReferenceAssembly>true</ProduceReferenceAssembly>
<Nullable>enable</Nullable>
</PropertyGroup>

**Reply by aritchie** [reply]
There is clearly something wrong with your setup.  The samples, the main repo, everything compiles just fine on all environments on my end.  

There is no netstandard 3.1 so that definitely isn't going to work.  At this point, this has gone way out of the way of Shiny questions so you'll have to do a bit more digging on your end.  Best of luck

**Comment by cdavidyoung** [comment]
I understand that everything compiles fine on your end.  I just thought you
would like to know that all is not well when someone else tries to build
your samples.  I am a seasoned programmer so there is no doubt I'll figure
it out.  Perhaps I'll be able to contribute a solution to the project.

Charles

**Reply by aritchie** [reply]
There is no problem with the projects or the Shiny repo, so there is nothing to contribute for this aspect of things.  There are many people who use it fine everywhere.  It has over 1million downloads on nuget, so if there was a major problem, there would be a lot more people making noise.  It also wouldn't build in the online CI system.  I'm not trying to be rude, but it is clear you are new to this sector of .NET, so I would really recommend learning about the xplat stuff first before diving into things like Shiny.  Shiny BLE (and to a degree beacons) are also built on other .NET practices like reactive (RX) that can be difficult if you are new to the ecosystem.

Best of luck

**Comment by cdavidyoung** [comment]
Thank you, Allan.  I am not trying to be rude either because it is obvious
that you and others have put much work into this project.  I have tried
other BLE plugins and this one seems to be the most promising.

I seem to have solved the problem by using the option Open with Visual
Studio on github.  Somehow downloading the zip file is not the same.  When
I compare the 2 Beacons folders I see lots of differences so perhaps the
zip file needs to be updated?

Charles

**Reply by aritchie** [reply]
The zip file has nothing to do with any of this.  GitHub simply wraps up the latest version online.  I am actively working on the samples, so depending on when you cloned it - there will be changes.
